src/sallyport/mod.rs

```diff
@@ -195,32 +197,95 @@ impl<'a> Cursor<'a> {
     pub unsafe fn alloc<T>(
         self,
         count: usize,
-    ) -> core::result::Result<(Cursor<'a>, &'a mut [T]), ()> {
+    ) -> core::result::Result<(Cursor<'a>, &'a mut [MaybeUninit<T>]), ()> {
         let mid = {
-            let (padding, data, _) = self.0.align_to_mut::<T>();
+            let (padding, data, _) = self.0.align_to_mut::<MaybeUninit<T>>();
 
             if data.len() < count {
                 return Err(());
             }
 
-            padding.len() + size_of::<T>() * count
+            padding.len() + size_of::<MaybeUninit<T>>() * count
         };
 
         let (data, next) = self.0.split_at_mut(mid);
 
-        Ok((Cursor(next), data.align_to_mut::<T>().1))
+        Ok((Cursor(next), data.align_to_mut::<MaybeUninit<T>>().1))
     }
 
-    /// Copies data from a value into a slice using self.alloc().
+    /// Copies data from a slice into a slice using self.alloc().
     #[allow(dead_code)]
-    pub fn copy_slice<T: Copy>(
+    pub fn copy_from_slice<T: 'a + Copy>(
         self,
-        value: &[T],
+        src: &[T],
     ) -> core::result::Result<(Cursor<'a>, &'a mut [T]), ()> {
-        let (c, slice) = unsafe { self.alloc(value.len())? };
-        slice.copy_from_slice(value);
+        let (c, slice) = unsafe { self.alloc::<T>(src.len())? };
+        let slice = unsafe { &mut *(slice as *mut [MaybeUninit<T>] as *mut [T]) };
+
+        // FIXME: should be volatile
+        slice.copy_from_slice(src);
+
         Ok((c, slice))
     }
+
+    /// Copies data into a slice from a slice using self.alloc().
+    #[allow(dead_code)]
```

Comment:
I don't think this is dead code.

Reply:
This is because of cargo and publishing `enarx-keepldr` on crates.io.
We therefore do not have `sallyport` as a dependency in `enarx-keepldr`, but `src/sallyport/mod.rs` and for the shims the `internal/sallyport` crate.
Although `src/sallyport/mod.rs` looks like a library crate, it is not and the compiler and clippy will warn, if any function is not used in the `enarx-keepldr`, which are most of it.